### PR TITLE
Move the whole kernel ELF to VMA

### PR DESCRIPTION
**Pull request by YanWQ-monad**
### Motivation

On the original kernel ELF layout, both `__kernel_start` and `__kernel_end` lay in the LMA, while the code was placed in the VMA. This might cause some obstacles in addressing on some code model, since they spans a long distance of address space (> 4 GiB). For example, RISC-V's default addressing mode can only access symbols lain in ±2 GiB window.

### Proposal

To address the issue, here move the whole kernel into VMA, including `__kernel_start` and `__kernel_end`. Linux also does it this way. Then, all the symbols are located in adjacent memory areas, which should solve the addressing-related issue.

### CODE REVIEW

**Review by junyang-zh — COMMENTED** [review]
I am OK with this PR. But you need to rebase after #807 is merged to rerun CI. And you should edit the aster frame revision to your commit hash in OSDK default dependency to make OSDK CI pass.

here:

https://github.com/asterinas/asterinas/blob/cd3faa8123678efc1290adfc7bbf6f2a2c1cfd1f/osdk/src/util.rs#L14-L17

**Review by junyang-zh — APPROVED** [review]
It looks good to me. Thanks for the contribution. Need final decision from @tatetian.

**Comment by tatetian** [comment]
@YanWQ-monad Th PR looks good. Could you rebase the PR to the latest main branch? It includes the new page table implementation.

File: framework/aster-frame/src/arch/x86/boot/boot.S
Comment (tatetian) [inline framework/aster-frame/src/arch/x86/boot/boot.S:20]: VMA = Virtual Memory Area? This is a Linux term. Asterinas does not have this concept. Could you give this address another name that is most fitting in the context of Asterinas?

Comment (YanWQ-monad) [reply framework/aster-frame/src/arch/x86/boot/boot.S:20]: It was just copied from `x86_64.ld` in fact.

What about `KERNEL_LOADED_OFFSET`? It's named after
https://github.com/asterinas/asterinas/blob/main/framework/aster-frame/src/vm/kspace.rs#L49-L51

Comment (tatetian) [reply framework/aster-frame/src/arch/x86/boot/boot.S:20]: I see. I guess VMA stands for virtual memory address. So it is acceptable, although this abbreviation is seldomly used in Asterinas. So it is not a particularly good abbreviation. Let's keep the name for now.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**
